### PR TITLE
Aligned to stock Java 11

### PR DESCRIPTION
**Pull request by sbluhm**
## What does this PR change?

- Don't use Java com.sun.xml.bind module if not available.
- Use Jaxb bundles if available.
- 
## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: No functional change.

- [X] **DONE**

## Test coverage
- No tests: Unsure how to test. Built successfully on CentOS8 and LEAP15.2.

- [X] **DONE**

## Links

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by mcalmer** [comment]
Please investigate how spacewalk-setup works. In share/ there is already a collection of "sed" scripts or perl regex scripts which change configurations in the system. Some also depend on criteria (e.g. tomcat version). I think using a sed to change a sed condition is not very useful :-)

Try to use the mechanism and add or change conditions as needed. I think we could also use this mechanism for taskomatic.
I am fine with removing --add-module from rhn_taskomatic_daemon.conf file and add this condition in spacewalk-setup to /etc/rhn/taskomatic.conf.

We only need to take care of the update case.

**Comment by sbluhm** [comment]
I am not sure the spacewalk-setup code can be re-used for that. As far as I understand, the setup/share files can only run regex and no system call (i.e. check for OS or java output).
So I would remove the --add-module option from tomcat.java_opts and add it from a new tomcat.java_opts.suse in the similar way to https://github.com/uyuni-project/uyuni/blob/ab53c20e6c3b5bddd137853a890481d84538c228/spacewalk/setup/spacewalk-setup.spec#L151
The sed after sed will remain untouched (not sure of the purpose really or if it will impact with my planned change that I just mentioned as it changes the order of --add-modules).

I am still unsure what to do with the taskomatic sed. I will look into this later.

File: java/buildconf/build-props.xml [hunk 1]
```diff
@@ -108,7 +127,7 @@
       value="bcel ${cglib-jar} commons-beanutils commons-cli commons-codec
              commons-collections commons-digester commons-discovery
              commons-el commons-fileupload commons-io ${commons-lang} commons-logging ${commons-validator} ${hibernate}
-             ${javamail} smtp jdom ${jmock-jars}
+	     ${javamail} smtp jdom ${jmock-jars}
```
Comment (cbosdo) [inline java/buildconf/build-props.xml:130]: Seems like a few spaces have mistakenly been replaced by a tab...

File: java/buildconf/build-props.xml [hunk 2]
```diff
@@ -180,7 +199,7 @@
              commons-digester commons-discovery commons-el commons-fileupload commons-io
              ${commons-lang} commons-logging
              ${commons-validator} concurrent dom4j ${hibernate} ${jta11-jars}
-             jaf ${jasper-jars} ${javamail} jcommon jdom ${other-jars}
+	     jaf ${jasper-jars} ${javamail} ${jaxb} jcommon jdom ${other-jars}
```
Comment (cbosdo) [inline java/buildconf/build-props.xml:202]: spacing issue here too

**Review by cbosdo — CHANGES_REQUESTED** [review]
Seems good to me... but I'm not a packaging and java module expert. Just need to fix two spacing issues though.

File: java/buildconf/build-props.xml (hunk 1 above)
Comment (sbluhm) [reply java/buildconf/build-props.xml]: Thank you, I couldn't figure out what was wrong with it! Didn't think of tabs.

**Comment by sbluhm** [comment]
@mcalmer , wasn't quite able to fulfil your requests. I only cleaned it up a little.
1. Taskomatic setup is currently not really in spacewalk-setup. So I kept my original changes as is.
2. Spacewalk-setup does not provide for feature detection in these scripts. Only regex. (or at least that is how I understand it). So for this, I followed the method in the spec file as written above.

Overall, I hope these are "temporary" changes until Leap has the required jaxb packages OOTB. Then we can skip the overall module logic and clean up.

If this approach is not OK, can you please give me more detailled instructions where to do what changes? I couldn't figure it out,

**Review by cbosdo — APPROVED** [review]
good for me, but I am surely not an expert in Java packaging

**Comment by mcalmer** [comment]
Let's go with this for now. Maybe we change it when we find a better solution.

**Pull request merged**